examples/rvc-app/rvc-common/src/rvc-device.cpp

```diff
@@ -91,9 +91,11 @@ void RvcDevice::HandleRvcCleanChangeToMode(uint8_t newMode, ModeBase::Commands::
 
 void RvcDevice::HandleOpStatePauseCallback(Clusters::OperationalState::GenericOperationalError & err)
 {
+    uint8_t tmpPausedState = mOperationalStateInstance.GetCurrentOperationalState();
```

Comment:
Would it make sense to instead have a `mResumeState` instead or remembering the pause?

This seems to make the "if pausing in X state, what is the Resume state" in a single place.
This is also visible in the "resume" call, however there it is mixed together with the resume that handles charging and docked.

```cpp
switch (mOperationalStateInstance.GetCurrentOperationalState()) {
    case to_underlying(RvcOperationalState::OperationalStateEnum::kSeekingCharger):
        mResumeState = RvcOperationalState::OperationalStateEnum::kSeekingCharger;
        break;
    default:
        mResumeState = OperationalState::OperationalStateEnum::kRunning;
        break
}
```

Reply:
The names of the variables are equivalent; `mPausedState == mResumeState`. It's a different English. Both seem fine to me however `mResumeState` seems to read better in the `HandleOpStateResumeCallback`.

Given that the `HandleOpStatePauseCallback` is only called if the device is in a pause-compatible state, i.e. `Running` or `SeekingCharger`, the switch statement can be replaced by `mResumeState = mOperationalStateInstance.GetCurrentOperationalState()`.

Also, thinking about it, given that the `HandleOpStatePauseCallback` is not called if the device is in the pause state, there is no need to check if the setting to paused state succeeded or failed before setting `mResumeState` since it is only used if the device is in the paused state. Hence we can remove the `tmpPausedState` variable and set `mResumeState` whether or not the setting to the paused state succeeds or fails.

I have made some changes here to reflect this.

Reply:
The difference in naming in my mind was:  
  - mPausedState == "the state that it was when we paused"
  - mResumeState == "the state to move into if resume is called".

They are not necesarely the same.

Reply:
The benefit in readability of using `ResuseState` is that you have less long casts towards `to_underlying`

Reply:
I agree with you about what the names mean. However, due to the fact that there are states that can be resumed but not paused, I don't think that it would be sensible to have a variable to keep track of the state that the device should be resumed to. If we did this, the logic would be scattered in every place were we enter a state that can be resumed.